modules/mobile/spec/request/letters_request_spec.rb

```diff
@@ -155,10 +245,32 @@
         end
       end
     end
+
+    context 'with an invalid letter type' do
+      it 'matches the letters schema' do
+        post '/mobile/v0/letters/not_real/download', headers: iam_headers
+
+        expect(response).to have_http_status(:internal_server_error)
+        expect(response.parsed_body).to eq(
+          {
+            'errors' => [
+              {
+                'title' => 'Invalid letter type',
+                'detail' => 'Invalid letter type',
+                'code' => '500',
+                'source' => 'Lighthouse::LettersGenerator::Service',
+                'status' => '500',
+                'meta' => { 'message' => 'Letter type of not_real is not one of the expected options' }
+              }
+            ]
+          }
+        )
+      end
+    end
   end
 
   describe 'Error Handling' do
```

Comment:
I think you could argue that these specs are a little light on handling the error cases. That's because the upstream team didn't give us any VCRs to work with. I recorded some responses from the staging rails terminal, but aside from 200 and 404 (tested further down in these specs), most error cases are difficult to reproduce and difficult to capture. So for error handling, I tested internally raised errors (like invalid letter type) and we have a test in this error handling section for "user not found".